practice/case.py

```diff
@@ -28,15 +28,50 @@ def detect_input_case(word):
     >>> detect_input_case('CONSTANT_CASE')
     'constant'
     """
-    if '-' in word:
+    if '-' in word and word == word.lower():
+        print(word, 'kebob')
         return 'kebob'
-    elif '_' in word:
-        if word[0].isupper():
-            return 'constant'
-        else:
-            return 'snake'
-    else:
+    elif '_' in word and word == word.upper():
+        print(word, 'constant')
+        return 'constant'
+    elif '_' in word and word == word.lower():
+        print(word, 'snake')
+        return 'snake'
+    elif word[0] == word[0].upper() and word != word.upper() and word.isalpha():
+        print(word, 'camel')
         return 'camel'
+    else:
+        raise Exception(word)
```

Comment:
I would raise `ValueError` because the input value is un-interpretable. It's fun to read through the built in exceptions and see which are most appropriate. https://docs.python.org/3.5/library/exceptions.html